internal/mode/static/state/dataplane/configuration.go

```diff
@@ -72,6 +75,44 @@ func buildSSLKeyPairs(
 	return keyPairs
 }
 
+func buildCertBundles(
+	configMaps map[types.NamespacedName]*graph.ConfigMap,
+	backendGroups []BackendGroup,
+) map[CertBundleID]CertBundle {
+	bundles := make(map[CertBundleID]CertBundle)
+	refByBG := make(map[CertBundleID]bool)
+
+	// We only need to build the cert bundles if there are valid backend groups that reference them.
+	if len(backendGroups) == 0 {
+		return bundles
+	}
+	for _, bg := range backendGroups {
+		if bg.Backends == nil {
+			continue
+		}
+		for _, b := range bg.Backends {
+			if !b.Valid || b.VerifyTLS == nil {
+				continue
+			}
+			refByBG[b.VerifyTLS.CertBundleID] = true
+		}
+	}
+
+	for cmName, cm := range configMaps {
+		id := generateCertBundleID(cmName)
+		if !refByBG[id] {
+			continue
+		}
+		if cm.Source.Data != nil || len(cm.Source.Data) > 0 {
```

Comment:
I wonder if we can move this logic into the graph package. For example, `ConfigMap` type can have the Bundle field populated with the Bundle or CA. Although we would probably need to rename ConfigMap to CAConfigMap.